ros/src/twist_controller/twist_controller.py

```diff
@@ -1,14 +1,52 @@
+from yaw_controller import YawController
+from lowpass import LowPassFilter
+from pid import PID
+
+import rospy
 
 GAS_DENSITY = 2.858
 ONE_MPH = 0.44704
 
 
 class Controller(object):
     def __init__(self, *args, **kwargs):
-        # TODO: Implement
-        pass
+        self.yaw_control = YawController(
+                kwargs['wheel_base'],
+                kwargs['steer_ratio'],
+                kwargs['min_speed'],
+                kwargs['max_lat_accel'],
+                kwargs['max_steer_angle'])
+
+        pid_gains = kwargs['throttle_gains']
+        self.pid = PID(pid_gains[0], pid_gains[1], pid_gains[2])
+        self.lpf = LowPassFilter(0.2,0.1)
+
+        self.t = None
+
+    def control(self, target, current):
```

Comment:
Can you add comments to this section?
What is the control algorithm?

Reply:
Will do